### ISSUE TITLE
ToggleSwitch: add support for p:confirm

### ISSUE THREAD

**Issue by AlessandroBori** [primefaces/primefaces#9983]
### Description

Hi,
is it possible a Confirmable behavior implementation on ToggleSwitch? maybe on Check box too..

### Describe the solution you would like

For Example it'll be great having a switch turning on an option ad asking "Are you sure?" and leave in off on negative answer

**Comment by tandraschko** [comment]
Did you try? this is not a forum, its a issue tracker

**Comment by AlessandroBori** [comment]
ToggleSwitch does not implements confirmable interface, so i need to do it manually 

I simply suggested a new feature.

**Comment by jepsar** [comment]
ToggleSwitch is not a UICommand.

**Comment by AlessandroBori** [comment]
That's the point i think. It'll be very useful indeed...

**Comment by melloware** [comment]
Yeah I don't think it can be a UIInput and a UICommand.  I don't see how this feature will be possible.

**Comment by tandraschko** [comment]
p:confirm is not really limited to UICommand per se

**Comment by fcorneli** [comment]
I'm trying to get my head around as to why `Confirmable` even exists. Maybe this could have been implemented without such additional interface.

What also would be nice, is that `p:confirm` could add:
```xml
<p:confirmDialog global="true">
            <p:commandButton value="No" type="button" styleClass="ui-confirmdialog-no"/>
            <p:commandButton value="Yes" type="button" styleClass="ui-confirmdialog-yes" />
        </p:confirmDialog>
```
itself to the view root in case none is present. Right now everybody is copy-pasting the above all over the place I guess.

**Comment by Rapster** [comment]
> I'm trying to get my head around as to why Confirmable even exists. Maybe this could have been implemented without such additional interface.

We all welcome new ideas to improve our base code, but maybe you should develop what makes you think that...

> Right now everybody is copy-pasting the above all over the place I guess.

Wrong guess, JSF has a powerful feature called templates, and most of your pages should "inherit" that template, therefore you can define common behavior like confirmDialog in that template, so it's only defined once 😉 (and of course `ConfirmDialog#global` has to be set)

**Comment by tandraschko** [comment]
Yep
Templating is the perfect place to add this

**Comment by melloware** [comment]
Yep its exactly what I do in my projects I declare this once for the whole app in the template.
